lib/galaxy/datatypes/binary.py

```diff
@@ -169,6 +171,136 @@ def _get_samtools_version( self ):
                 break
         return version
 
+    def split(cls, input_datasets, subdir_generator_function, split_params):
+
+        # 1) by_rname -> splits the bam into files based on the chromosome
+        # 2) by_interval -> splits the bam into files based on  a defined bp length, and does so across the entire genome present in the BAM file
+        # 3) by_read -> splits the bam into files based on the number of reads encountered (if multiple files, all other files match the interval as the first)
+
+        if split_params is None:
+            return
+        if len(input_datasets) > 1:
+            raise Exception("BAM file splitting does not support multiple files")
```

Comment:
for example this:
if len(input_datasets) > 1:
            raise ValueError("BAM file splitting does not support multiple files")
?